api/proto/metrics/metrics.proto

```diff
@@ -468,6 +468,41 @@ message ZMetricVolume {
   uint64 freeBytes = 9;
 }
 
+// Metrics for a snapshot
+// When a snapshot is created, its disk space is initially shared between
+// the snapshot and the file system, and possibly with previous snapshots.
+// As the file system changes, disk space that was previously shared becomes
+// unique to the snapshot, and thus is counted in the snapshot's used property.
+// Additionally, deleting snapshots can increase the amount of disk space
+// unique to (and thus used by) other snapshots.
+message ZMetricSnapshot {
+  // Snapshot UUID
+  string uuid = 1;
+  // User-friendly name on controller
+  string display_name = 2;
+  // Identifies the amount of space consumed by the dataset and all its
+  // descendants. (in byte)
+  uint64 used_space = 3;
+  // Identifies the amount of data accessible by this snapshot, which might
+  // or might not be shared with other datasets in the pool. When a snapshot or
+  // clone is created,  It initially references the same amount of space as the
+  // file system or snapshot it was created from because
+  // its contents are identical. (in byte)
+  uint64 referenced = 4;
+  // Identifies the compression ratio achieved for this snapshot,
+  // expressed as a multiplier.
+  double compressratio = 5;
+  // Specifies the logical size of the volume. (in byte)
+  uint64 vol_size = 6;
```

Comment:
Can this be different than the size of the volume/parent of the snapshot?

Reply:
No, this is the size of the logical volume that this snapshot belongs to

Reply:
How about adding that to the comment?

Reply:
As I understand from my experiments this field equals with max volume size and potentially will help us in future if we will decide to support resize of volumes. Is it correct?